libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CloudAdapter.cs

```diff
@@ -154,19 +154,23 @@ public async Task ConnectNamedPipeAsync(string pipeName, IBot bot, string appId,
             var connectionId = Guid.NewGuid();
             using (var scope = Logger.BeginScope(connectionId))
             {
+                do
+                {
 #pragma warning disable CA2000 // Dispose objects before losing scope: StreamingRequestHandler is responsible for disposing StreamingConnection
-                var connection = new NamedPipeStreamingConnection(pipeName, Logger);
+                    var connection = new NamedPipeStreamingConnection(pipeName, Logger);
 #pragma warning restore CA2000 // Dispose objects before losing scope
 
-                using (var streamingActivityProcessor = new StreamingActivityProcessor(authenticationRequestResult, connection, this, bot))
-                {
-                    // Start receiving activities on the named pipe
-                    _streamingConnections.TryAdd(connectionId, streamingActivityProcessor);
-                    Log.WebSocketConnectionStarted(Logger);
-                    await streamingActivityProcessor.ListenAsync(CancellationToken.None).ConfigureAwait(false);
-                    _streamingConnections.TryRemove(connectionId, out _);
-                    Log.WebSocketConnectionCompleted(Logger);
+                    using (var streamingActivityProcessor = new StreamingActivityProcessor(authenticationRequestResult, connection, this, bot))
+                    {
+                        // Start receiving activities on the named pipe
+                        _streamingConnections.TryAdd(connectionId, streamingActivityProcessor);
+                        Log.WebSocketConnectionStarted(Logger);
+                        await streamingActivityProcessor.ListenAsync(CancellationToken.None).ConfigureAwait(false);
+                        _streamingConnections.TryRemove(connectionId, out _);
+                        Log.WebSocketConnectionCompleted(Logger);
+                    }
                 }
+                while (_streamingConnections.IsEmpty);
```

Comment:
_streaming Connections will have both named pipe and web socket connections. So, it may not be empty if there are any web socket connections active. It would be better to log that named pipe got disconnected and just reconnect.

Reply:
Hi @msomanathan, we changed the loop to log and reconnect without condition.
Should we add a _Cancellation Token_ parameter to the _ConnectNamedPipeAsync_ method to use it to break the loop in case it's needed?